src/main/java/sx/blah/discord/util/LongMapCollector.java

```diff
@@ -0,0 +1,56 @@
+package sx.blah.discord.util;
+
+import sx.blah.discord.handle.obj.IIDLinkedObject;
+import sx.blah.discord.util.cache.LongMap;
+
+import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collector;
+
+public class LongMapCollector<T extends IIDLinkedObject> implements Collector<T, List<T>, LongMap<T>> {
+
+	private LongMapCollector() {
+
+	}
+
+	@Override
+	public Supplier<List<T>> supplier() {
+		return ArrayList::new;
```

Comment:
A non-concurrent collector may be used in the context of parallelStream() given that supplier, accumulator, and combiners are serially thread-confined. It is unnecessary to build in additional synchronization for our purposes.